js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillRunsTab.tsx

```diff
@@ -0,0 +1,187 @@
+import {StickyTableContainer} from '../../ui/StickyTableContainer';
+
+export const BackfillRunsTab = ({backfill}: {backfill: BackfillDetailsBackfillFragment}) => {
+  const [view, setView] = useQueryPersistedState<'timeline' | 'list'>({
+    defaults: {view: 'timeline'},
+    queryKey: 'view',
+  });
+  const filter: RunsFilter = useMemo(
+    () => ({tags: [{key: DagsterTag.Backfill, value: backfill.id}]}),
+    [backfill],
+  );
+
+  const actionBarComponents = (
+    <>
+      <ButtonGroup
+        activeItems={new Set([view])}
+        onClick={(id: 'timeline' | 'list') => {
+          setView(id);
+        }}
+        buttons={[
+          {id: 'timeline', icon: 'gantt_waterfall', label: 'Timeline'},
+          {id: 'list', icon: 'list', label: 'List'},
+        ]}
+      />
+    </>
+  );
+
+  return (
+    <Box flex={{direction: 'column'}} style={{minHeight: 0}}>
+      {view === 'timeline' ? (
+        <ExecutionRunTimeline
+          filter={filter}
+          startTimestamp={backfill.timestamp}
+          endTimestamp={backfill.endTimestamp}
+          actionBarComponents={actionBarComponents}
+        />
+      ) : (
+        <ExecutionRunTable filter={filter} actionBarComponents={actionBarComponents} />
+      )}
+    </Box>
+  );
+};
+
+const ExecutionRunTable = ({
+  filter,
+  actionBarComponents,
+}: {
+  filter: RunsFilter;
+  actionBarComponents: React.ReactNode;
+}) => {
+  const {queryResult, paginationProps} = usePaginatedRunsTableRuns(filter);
+  const pipelineRunsOrError = queryResult.data?.pipelineRunsOrError;
+
+  useQueryRefreshAtInterval(queryResult, 15000);
+
+  if (!pipelineRunsOrError) {
+    return (
+      <Box padding={{vertical: 48}}>
+        <Spinner purpose="page" />
+      </Box>
+    );
+  }
+  if (pipelineRunsOrError.__typename !== 'Runs') {
+    return (
+      <Box padding={{vertical: 64}}>
+        <NonIdealState icon="error" title="Query Error" description={pipelineRunsOrError.message} />
+      </Box>
+    );
+  }
+
+  return (
+    <StickyTableContainer $top={0}>
+      <RunTable runs={pipelineRunsOrError.results} actionBarComponents={actionBarComponents} />
+      {pipelineRunsOrError.results.length > 0 ? (
+        <div style={{marginTop: '16px'}}>
+          <CursorHistoryControls {...paginationProps} />
+        </div>
+      ) : null}
+    </StickyTableContainer>
+  );
+};
+
+const POLL_INTERVAL = 60 * 1000;
+const ONE_WEEK = 7 * 24 * 60 * 60 * 1000;
+
+const ExecutionRunTimeline = ({
+  filter,
+  startTimestamp,
+  actionBarComponents,
+  endTimestamp,
+}: {
+  filter: RunsFilter;
+  actionBarComponents: React.ReactNode;
+  startTimestamp: number;
+  endTimestamp: number | null;
+}) => {
+  const timelineRange: [number, number] = useMemo(() => {
+    return [startTimestamp * 1000, endTimestamp ? endTimestamp * 1000 : Date.now() + ONE_WEEK];
+  }, [startTimestamp, endTimestamp]);
+
+  const runsForTimelineRet = useRunsForTimeline(timelineRange, filter);
```

Comment:
Hmm the indexeddb caching doesnt take filtering into account. Think we could add that here? Or maybe we just skip the indexeddb caching entirely for this page?

Reply:
Ooh interesting that's good to know, I think that explains what I was seeing!